src/nvme/types.h

```diff
@@ -9007,4 +9025,350 @@ struct nvme_ns_mgmt_host_sw_specified {
 };
 #endif /* SWIG */
 
+/**
+ * enum nvme_lm_cdq_fields - Controller Data Queue command fields
+ *
+ * @NVME_LM_CREATE_CDQ:					Create CDQ select option
+ * @NVME_LM_DELETE_CDQ:					Delete CDQ select option
+ * @NVME_LM_CDQ_SELECT_SHIFT:				Shift to set Select (SEL) field
+ * @NVME_LM_CDQ_SELECT_MASK:				Mask to set SEL field
+ * @NVME_LM_QUEUE_TYPE_USER_DATA_MIGRATION_QUEUE:	User Data Migration Queue type
+ * @NVME_LM_QUEUE_TYPE_SHIFT:				Shift amount to set Queue Type (QT) field
+ * @NVME_LM_QUEUE_TYPE_MASK:				Mask to set QT field
+ * @NVME_LM_CREATE_CDQ_PC:				Physically Contiguous
+ * @NVME_LM_CREATE_CDQ_CNTLID_SHIFT:			Shift amount to set CNTLID field
+ * @NVME_LM_CREATE_CDQ_CNTLID_MASK:			Mask to set CNTLID field
+ * @NVME_LM_CREATE_CDQ_CDQID_SHIFT:			Shift amount to get CDQID field from Create
+ *							response in completion dword0
+ * @NVME_LM_CREATE_CDQ_CDQID_MASK:			Mask to get CNTLID field from Create
+ *							response in completion dword0
+ * @NVME_LM_DELETE_CDQ_CDQID_SHIFT:			Shift amount to set CDQID field for deletion
+ * @NVME_LM_DELETE_CDQ_CDQID_MASK:			Mask to set CDQID field for deletion
+ */
+enum nvme_lm_cdq_fields {
+	NVME_LM_CREATE_CDQ				= 0,
+	NVME_LM_DELETE_CDQ				= 1,
+	NVME_LM_CDQ_SELECT_SHIFT			= 0,
+	NVME_LM_CDQ_SELECT_MASK				= 16,
```

Comment:
`0xffff`?

Reply:
Also please use the `SEL` instead the `Select`. This makes it consistent with the recently added field names (and usually better searchable in the spec).  Also is there a reason you don't add the `MOS` (looking at the 2.1 spec)